javav2/example_code/s3/src/main/java/com/example/s3/scenario/S3Actions.java

```diff
@@ -0,0 +1,376 @@
+                UploadPartRequest uploadPartRequest1 = UploadPartRequest.builder()
+                    .bucket(bucketName)
+                    .key(key)
+                    .uploadId(uploadId)
+                    .partNumber(1)
+                    .contentLength((long) (5 * mB)) // Specify the content length
+                    .build();
+
+                CompletableFuture<CompletedPart> part1Future = getAsyncClient().uploadPart(uploadPartRequest1,
+                        AsyncRequestBody.fromByteBuffer(getRandomByteBuffer(5 * mB)))
+                    .thenApply(uploadPartResponse -> CompletedPart.builder()
+                        .partNumber(1)
+                        .eTag(uploadPartResponse.eTag())
+                        .build());
+
+                // Upload part 2.
+                UploadPartRequest uploadPartRequest2 = UploadPartRequest.builder()
+                    .bucket(bucketName)
+                    .key(key)
+                    .uploadId(uploadId)
+                    .partNumber(2)
+                    .contentLength((long) (3 * mB))
+                    .build();
+
+                CompletableFuture<CompletedPart> part2Future = getAsyncClient().uploadPart(uploadPartRequest2,
+                        AsyncRequestBody.fromByteBuffer(getRandomByteBuffer(3 * mB)))
+                    .thenApply(uploadPartResponse -> CompletedPart.builder()
+                        .partNumber(2)
+                        .eTag(uploadPartResponse.eTag())
+                        .build());
+
+                // Combine the results of both parts.
+                return CompletableFuture.allOf(part1Future, part2Future)
+                    .thenCompose(v -> {
+                        CompletedPart part1 = part1Future.join();
+                        CompletedPart part2 = part2Future.join();
+
+                        CompletedMultipartUpload completedMultipartUpload = CompletedMultipartUpload.builder()
+                            .parts(part1, part2)
+                            .build();
+
+                        CompleteMultipartUploadRequest completeMultipartUploadRequest = CompleteMultipartUploadRequest.builder()
+                            .bucket(bucketName)
+                            .key(key)
+                            .uploadId(uploadId)
+                            .multipartUpload(completedMultipartUpload)
+                            .build();
+
+                        // Complete the multipart upload
+                        return getAsyncClient().completeMultipartUpload(completeMultipartUploadRequest);
+                    });
+            })
+            .thenAccept(response -> System.out.println("Multipart upload completed successfully"))
+            .exceptionally(ex -> {
+                System.err.println("Failed to complete multipart upload: " + ex.getMessage());
+                throw new RuntimeException(ex);
+            });
+    }
+
+    // snippet-start:[s3.java2.delete_objects.main]
```

Comment:
This is not the snippet in the metadata. The deleteObjects in the code library is not async.

Reply:
Nice pickup